design/automate_release_design.md

```diff
@@ -0,0 +1,204 @@
+| [Github actions](https://docs.github.com/en/free-pro-team@latest/actions) | CI/CD platform that is event-driven to help automate task and execute workflows in repositories |
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+Release documentation
+
+- [Secrets Provider release documentation](https://github.com/cyberark/secrets-provider-for-k8s/blob/master/CONTRIBUTING.md#releases)
+
+Open issues:
+
+- [Secrets Provider - Release automation](https://app.zenhub.com/workspaces/palmtree-5d99d900491c060001c85cba/issues/cyberark/secrets-provider-for-k8s/233)
+
+- [Secrets Provider - Github action for dispatching events](https://github.com/cyberark/secrets-provider-for-k8s/issues/239)
+
+- [Community - Guidelines and tools for releasing artifacts to separate suite components](https://github.com/cyberark/community/issues/77)
+
+PR:
+
+- https://github.com/cyberark/secrets-provider-for-k8s/pulls
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+The Secrets Provider release process has become a long and heavily manual process. This has become more evident with the introduction of Helm to the project which now adds additional steps to the release process. 
+
+Goals for this effort:
+
+1. Automate the parts we can.
+2. Knowledge sharing by providing guidelines for event-driven actions across multiple repositories.
+3. Push tag on every master build for easier testing and to better align ourselves with other open source projects
+
+## Solution
+
+[//]: # "Elaborate on the solution you are suggesting in this page. Address the functional requirements and the non functional requirements that this solution is addressing. If there are a few options considered for the solution, mention them and explain why the actual solution was chosen over them. Add an execution plan when relevant. It doesn't have to be a full breakdown of the feature, but just a recommendation to how the solution should be approached."
+
+As part of our Secrets Provider for K8s [release process](https://github.com/cyberark/secrets-provider-for-k8s/blob/master/CONTRIBUTING.md#releases) we need to manually package our Secrets Provider Helm Chart and open a PR attaching the artifact in our [cyberark/helm-charts](cyberark/helm-charts) repository. This process can be automated by introducing Github actions into our workflows and will be addressed in the below solution.
+
+The solution will be twofold and impact both the `cyberark/secrets-provider-for-k8s` Jenkins pipeline and `cyberark/helm-chart` repository. The flow at a high-level is as follows:
+
+`cyberark/secrets-provider-for-k8s` Jenkins pipeline
+
+1. Package the Secrets Provider Helm chart as part of Jenkins pipeline
+2. Dispatch an `version-release` event by sending a curl request with Helm package as part of request
```

Comment:
an -> a

Reply:
will the secrets-provider pipeline know if the `cyberark/helm-charts` failed its flow?

Reply:
I don't think it is smart enough yet to know but this can be something that can be explored. Isn't it up to the permission who is reviewing to approval or fail?

If the helm-chart repo has a failure what do you think the logical action would be? Just failing the tagged release build?